typed-racket-lib/typed-racket/typecheck/tc-expr-unit.rkt

```diff
@@ -352,6 +352,25 @@
           #:stx form
           "expected single value, got multiple (or zero) values")]))
 
+;; Apply `single-value` to a list of forms in order, accumulate the prop info
+(define (map-single-value forms [props '()])
+  (define any-res (-tc-any-results #f))
+  (let loop ((forms forms)
+             (props props))
+    (match forms
+     ['()
+      '()]
+     [(cons e rst)
+      (define tcr
+        (with-lexical-env+props
+          props
```

Comment:
I wonder if `with-lexical-env+props` is right here and not `with-lexical-env+props-simple`? The latter seems to do a little less work... TBH I'm not sure which to favor here.

Reply:
I used `with-lexical-env+props` because that's what `tc-body/check` uses for function bodies.